### PR TITLE
Move accelerate launch args parsing

### PR DESCRIPTION
**Pull request by tharapalanivel**
### Description of the change

Refactor accelerate launch args parsing logic out into `build/utils.py` and add unit tests

### Was the PR tested

- [x] I have added >=1 unit test(s) for every new method I have added.
- [x] I have ensured all unit tests pass

### CODE REVIEW

File: tests/build/test_utils.py [hunk 1]
```diff
@@ -78,3 +79,37 @@ def test_process_launch_training_args_peft_method(job_config):
...
+def test_process_accelerate_launch_args(job_config):
+    job_config_copy = copy.deepcopy(job_config)
+    args = process_accelerate_launch_args(job_config_copy)
+    assert args.config_file == "fixtures/accelerate_fsdp_defaults.yaml"
+    assert args.use_fsdp == True
+    assert args.tpu_use_cluster == False
+
+
+@patch("os.path.exists")
+def test_process_accelerate_launch_custom_fsdp(patch_path_exisits):
```
Comment (Ssukriti) [inline tests/build/test_utils.py:93]:
```suggestion
def test_process_accelerate_launch_custom_fsdp(patch_path_exists):
```

File: tests/build/test_utils.py [hunk 2]
```diff
@@ -78,3 +79,37 @@ def test_process_launch_training_args_peft_method(job_config):
...
+    patch_path_exisits.return_value = True
```
Comment (Ssukriti) [inline tests/build/test_utils.py:94]:
```suggestion
    patch_path_exists.return_value = True
```

**Review by Ssukriti — DISMISSED** [review]
Took a very quick look. The refactor looks good to me , thank you! I assume all the functionality is retained - that part I didnt check line by line, but we can test final image before delivery! Thank you!

**Review by anhuong — APPROVED** [review]
LGTM thanks so much Thara! Really great refactor and addition of tests! Thanks for cleaning things up!

**Pull request merged**